internal/queryexecutor/range_query_planner.go

```diff
@@ -36,15 +71,24 @@ func createQueryPlan(attribute string, conds *attributeTypeAndConditions) (*rang
 		},
 	}
 
+	if len(excludeKeys) > 0 && len(conds.conditions) == 0 {
+		// full scan with a seek to not equal to
+		p.startKey.ValuePosition = stateindex.Beginning
+		p.excludeKeys = excludeKeys
+		p.endKey.ValuePosition = stateindex.Ending
+
+		return p, nil
+	}
+
 	if len(conds.conditions) == 1 {
 		for c, v := range conds.conditions {
 			setPlanForSingleCondition(c, v, p)
-			return p, nil
 		}
+	} else {
+		setPlanForMultipleConditions(conds, p)
 	}
 
-	setPlanForMultipleConditions(conds, p)
-
+	p.excludeKeys = excludeKeys
```

Comment:
As an optimization, after setting the start and end key, you could prune the excludeKeys for values out side that range?
This does not affect correctness as the executor will never match the value of entries from within the range to an exlude-key from out side it... but, it might save some time. 

In addition, I am not sure I've seen a test for a range [x,y] with exclude keys outside it?

Reply:
Will be addressed in one of the following PRs.